src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountDecider.fs

```diff
@@ -0,0 +1,80 @@
+﻿module Deciders.BankAccountDecider
+
+open System
+open BankAccount
+open BankAccountPrimitives
+
+type OpenBankAccount =
+    { BankAccountId: AccountId
+      AccountNumber: AccountNumber
+      ClientId: ClientId
+      CurrencyISOCode: CurrencyCode
+      Now: DateTimeOffset }
+
+type RecordDeposit = { Amount: decimal; CashierId: CashierId; Now: DateTimeOffset }
+type WithdrawCashFromATM = { Amount: decimal; AtmId: AtmId; Now: DateTimeOffset }
+type CloseBankAccount = { Reason: string; Now: DateTimeOffset }
+
+type Command =
+    | OpenBankAccount of OpenBankAccount
+    | RecordDeposit of RecordDeposit
+    | WithdrawCashFromATM of WithdrawCashFromATM
+    | CloseBankAccount of CloseBankAccount
+
+let openBankAccount (command: OpenBankAccount) (bankAccount: BankAccount) =
+    match bankAccount with
+    | Open _ -> invalidOp "Account is already opened!"
+    | Closed _ -> invalidOp "Account is already closed!"
+    | NotInitialised _ ->
+        BankAccountOpened
+            { BankAccountId = command.BankAccountId
+              AccountNumber = command.AccountNumber
+              ClientId = command.ClientId
+              CurrencyISOCode = command.CurrencyISOCode
+              CreatedAt = command.Now
+              Version = 1 }
+
+let recordDeposit (command: RecordDeposit) (bankAccount: BankAccount) =
+    match bankAccount with
+    | NotInitialised _ -> invalidOp "Account is not opened!"
+    | Closed _ -> invalidOp "Account is closed!"
+    | Open openBankAccount ->
+        DepositRecorded
+            { BankAccountId = openBankAccount.Id
+              Amount = command.Amount
+              CashierId = command.CashierId
+              RecordedAt  = command.Now
+              Version = openBankAccount.Version + 1L }
+
+let withdrawCashFromATM (command: WithdrawCashFromATM) (bankAccount: BankAccount) =
+    match bankAccount with
+    | NotInitialised _ -> invalidOp "Account is not opened!"
+    | Closed _ -> invalidOp "Account is closed!"
+    | Open openBankAccount ->
```

Comment:
consider replacing `openBankAccount` with `x` - if you keep functions and expressions short, `x` can be "the thing we're working on". And if it's not obvious, thats a hint to extract to a function

Reply:
Or even `value`.

Reply:
Yes, `value` or something that names the role is definitely an excellent default
Using short names like `x` is definitely a subjective thing with downsides.

Reply:
I appreciate the functional flavour, but I prefer to keep the longer version. I understand the reasoning; it may happen that in the future, I'll change my mind, but for now, I don't like the single-letter naming. Especially the one I saw in Haskell code was making it too esoteric per my taste 😅 I know this may not be following the F# best practices, but I prefer to keep explicitness here.

Reply:
There is no official F# best practice on this

There are people who try to write Haskell in F#, and they definitely would use `x`. I'm not suggesting it from that angle of zealotry - I just happen to find it makes code regular and easy to read if you have standard cues and patterns in layout and naming.

I'd equally use the name `value` in C# too on the basis that it's a role name - explained better in https://blog.ploeh.dk/2020/11/30/name-by-role/